src/share/poudriere/jail.sh

```diff
@@ -505,6 +516,95 @@ build_native_xtools() {
 	BUILT_NATIVE_XTOOLS=1
 }
 
+install_from_ports() {
+	# Create our work and usr directories for later usage
+	mkdir ${JAILMNT}/work
+	mkdir ${JAILMNT}/usr
+
+	# Create our log location
+	LOGDIR="${POUDRIERE_DATA}/logs/base-ports"
+	if [ ! -d "${LOGDIR}" ] ; then
+		mkdir -p ${LOGDIR}
+	fi
+
+	# Figure out where finished OS packages need to be stashed
+	local PACKAGES=${JAILMNT}/.packages/
+	if [ ! -d "${PACKAGES}" ]; then
+		mkdir -p ${PACKAGES}
+	fi
+
+	# Create package for the system sources from the os/src port
+	msg "Building src: ${LOGDIR}/${JAILNAME}-buildsrc.log"
+	make -C ${PORTS_BASE}/os/src WRKDIR=${JAILMNT}/work/src BATCH=yes package >${LOGDIR}/${JAILNAME}-buildsrc.log 2>&1
+	if [ $? -ne 0 ] ; then
+		err 1 "Failed building src"
+	fi
+
+	# Install the package
+	local PKGFILE="${JAILMNT}/work/src/pkg/$(make PORTSDIR=${PORTS_BASE} -C ${PORTS_BASE}/os/src -V PKGNAME).txz"
+	local ABISTRING="$(make PORTSDIR=${PORTS_BASE} -C ${PORTS_BASE}/os/src -V PKG_ABISTRING)"
+	pkg-static -o ABI=${ABISTRING} -r ${JAILMNT} add ${PKGFILE}
+	if [ $? -ne 0 ] ; then
```

Comment:
This style of checking `$?` isn't right in Poudriere, nor needed.
`if ! pkg-static ...; then
    err
fi
`
or simply
`pkg-static ... || err`

Reply:
It's also wrong in Poudriere because the use of `set -e` will never make it to the `if [ $?` line. `set -e` might be off in this file but that's already a bug.